docs/config/index.md

```diff
@@ -366,18 +366,45 @@ export default defineConfig({
 })
 ```
 
-To get TypeScript working with the global APIs, add `vitest/globals` to the `types` field in your `tsconfig.json`
+#### TypeScript integration
 
-```json
-// tsconfig.json
-{
-  "compilerOptions": {
-    "types": ["vitest/globals"]
-  }
-}
-```
+To register global API types, follow one of the following methods:
+
+1. Use a `@types/vitest.d.ts` file
+
+    Create a `@types/vitest.d.ts` file in your project root that imports `vitest/globals`:
+
+    ```ts
+    import 'vitest/globals'
+    ```
+
+2. Type-import the global types from a source file
+
+    Add a type-import from `vitest/globals` in any `.ts` file (e.g., your [setup file](#setupfiles)). This registers the global APIs types automatically. Ensure the file is included in your `tsconfig.json` so TypeScript knows it's apart of the same project.
+
+    ```ts
+    // test-setup.ts
+    import type {} from 'vitest/globals'
+    ```
+
+3. Configure it in [`tsconfig.json#compilerOptions.types`](https://www.typescriptlang.org/tsconfig/#types)
+
+    Alternatively, add `vitest/globals` to the `types` field in `tsconfig.json`.
+
+    Warning: This disables TypeScript's automatic type detection, which could break your setup if your project relies on it. If you encounter types missing from other packages, you'll now need to manually register them here as well.
```

Comment:
I think we should not even include this method as many beginners have been following this without understanding how disruptive it is and makes TS harder for them to work with.

Reply:
Vitepress has a `::: danger` block we can use here